awx/main/tasks.py

```diff
@@ -1866,6 +1866,20 @@ def build_env(self, inventory_update, **kwargs):
             for env_k in inventory_update.source_vars_dict:
                 if str(env_k) not in env and str(env_k) not in settings.INV_ENV_VARIABLE_BLACKLIST:
                     env[str(env_k)] = unicode(inventory_update.source_vars_dict[env_k])
+            if inventory_update.plugin_path:
+                if inventory_update.plugin_path.startswith('@'):
+                    if inventory_update.source != 'scm':
+                        raise Exception('Direct paths to inventory plugins only usable with SCM sources.')
+                    plugin_path = os.path.join(kwargs.get('private_data_dir', ''), inventory_update.plugin_path[1:])
+                    env['ANSIBLE_INVENTORY_PLUGINS'] = os.path.dirname(plugin_path)
+                    plugin_name = os.path.basename(plugin_path).split('.')[0]
```

Comment:
Given  `@plugin/open.stack.v2.py` would result in `plugin_name = open`

I prefer something like `filename, file_extension = os.path.splitext('/path/to/somefile.ext')`

Reply:
Expectation in that case isn't clear. One reasonable guess might be `open.stack.v2`.

```
>>> plugin_path = '@plugin/open.stack.v2.py'
>>> os.path.splitext(plugin_path)
('@plugin/open.stack.v2', '.py')
```

so I think we'd prefer (setting aside whether any of this is a good idea)

```
>>> os.path.splitext(os.path.basename(plugin_path))[0]
'open.stack.v2'
```

But alas, this probably won't work. Here's an example I've been meaning to try:

https://github.com/AlanCoding/Ansible-inventory-file-examples/blob/master/plugins/name_conflict/alan.py

File named `alan.py` has declared `NAME = 'steve'` on the Ansible inventory plugin class.

So we could still infer the name of the plugin if we `exec` the file. But how can we do that safely?